### PR TITLE
Validate config to avoid wrong SN queries and enhance doc

### PR DESCRIPTION
**Pull request by louisfelix**
- Validate config to avoid wrong SN queries (get all incidents)
- enhance doc for new comers

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -59,6 +59,28 @@ type JSONResponse struct {
...
+	if len(c.ServiceNow.Password) == 0 {
+		errs.WriteString("password is missing\n")
+	}
+	if len(c.Workflow.IncidentGroupKeyField) == 0 {
+		errs.WriteString("incident_group_key_field is missing\n")
+	}
+
+	if errs.Len() > 0 {
+		return errors.New("Config file is invalid\n" + errs.String())
+	}
```
Comment (rolandvarga) [inline main.go]: Is there a way to append to the error object itself and return that wether it's `nil` or not?

File: main.go [hunk 2]
```diff
@@ -142,6 +164,11 @@ func loadConfigContent(configData []byte) (Config, error) {
 		return config, errYAML
 	}
 
+	errConfig := config.validate()
```
Comment (rolandvarga) [inline main.go]: I prefer using `err` without any suffixes as the value stays `nil` until an error is caught, but it's entirely subjective

File: main_test.go [hunk 3]
```diff
@@ -332,33 +333,69 @@ func TestApplyIncidentTemplate_Range(t *testing.T) {
...
-		Workflow:        WorkflowConfig{},
+		Workflow: WorkflowConfig{
+			IncidentGroupKeyField: "u_other_reference_1",
+			NoUpdateStates:        []json.Number{"6", "7"},
+			IncidentUpdateFields:  []string{"comments"},
+		},
 		DefaultIncident: defaultIncident,
 	}
-	config, err := loadConfigContent([]byte(configFile))
+	loadedConfig, err := loadConfigContent([]byte(configFile))
```
Comment (rolandvarga) [inline main_test.go]: Idiomatic testing usually follows the `want` & `got` naming of variables

Comment (louisfelix) [reply main_test.go]: @rolandvarga would you call the variable `want` or `wantedConfig` here?

File: main.go (hunk 1 above)
Comment (louisfelix) [reply main.go]: Hum, maybe, but I like the fact of not creating the `error` unless there is actually an error

File: main.go (hunk 2 above)
Comment (louisfelix) [reply main.go]: I will change it

File: main_test.go (hunk 3 above)
Comment (louisfelix) [reply main_test.go]: I changed it

File: main.go (hunk 1 above)
Comment (rolandvarga) [reply main.go]: I would argue that your function signature indicates you are returning an error, wether it's content is an error message or not is a different question and can be argued either way. With all that said LGTM!

**Pull request merged**